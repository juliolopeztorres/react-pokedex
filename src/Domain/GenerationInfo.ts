
const generationsInfo: {[key:string]: {offset: number, limit: number}} = {
  '1st': {
    offset: 0,
    limit: 151
  },
  '2nd': {
    offset: 151,
    limit: 100
  },
  '3rd': {
    offset: 251,
    limit: 135
  },
  '4th': {
    offset: 386,
    limit: 107
  },
  '5th': {
    offset: 493,
    limit: 156
  },
  '6th': {
    offset: 649,
    limit: 72
  },
  '7th': {
    offset: 721,
    limit: 88
  },
  '8th': {
    offset: 809,
    limit: 96
  },
  '9th': {
    offset: 905,
    limit: 111
  },
}

export default generationsInfo
