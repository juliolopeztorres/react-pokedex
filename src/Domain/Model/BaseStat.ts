
export default class BaseStat {

  constructor(public name : string, public value : number) {}
}
