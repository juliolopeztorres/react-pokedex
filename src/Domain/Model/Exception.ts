
export default class Exception {

  public static create(message: string, context?: object): Exception {
    return new Exception(message, context)
  }

  private constructor(
    public message: string,
    public context?: object
  ) {
  }
}
