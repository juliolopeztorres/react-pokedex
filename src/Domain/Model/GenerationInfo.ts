
class GenerationInfo {
  constructor(
    public offset: number,
    public limit: number
  ) {}
}

const generationsInfo = {
  '1st': new GenerationInfo(0, 151),
  '2nd': new GenerationInfo(151, 100),
  '3rd': new GenerationInfo(251, 135),
  '4th': new GenerationInfo(386, 107),
  '5th': new GenerationInfo(493, 156),
  '6th': new GenerationInfo(649, 72),
  '7th': new GenerationInfo(721, 88),
  '8th': new GenerationInfo(809, 96),
  '9th': new GenerationInfo(905, 111),
} as const

type Generation = keyof typeof generationsInfo

const generationLabels: Generation[] = Object.keys(generationsInfo) as Generation[]

export default generationsInfo
export { generationLabels, Generation }
