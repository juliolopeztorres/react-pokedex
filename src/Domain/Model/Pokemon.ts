export default class Pokemon {

  constructor(
    public id : number,
    public name : string
  ) {}
}
