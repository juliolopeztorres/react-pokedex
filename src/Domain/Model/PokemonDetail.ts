import BaseStat from "./BaseStat";

export default class PokemonDetail {

  constructor(
    public id : number,
    public name : string,
    public image : string,
    public types : string[],
    public backSprites : string[],
    public baseStats : BaseStat[]
  ) {}
}
