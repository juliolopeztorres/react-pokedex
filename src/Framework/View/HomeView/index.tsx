import React, { useCallback, useContext, useState } from "react";
import { hot } from "react-hot-loader/root";
import PokemonList from "./PokemonList";
import PokemonModal from "./PokemonModal";
import PokemonDetail from "./PokemonDetail";
import LoadingContainer from "../Component/LoadingContainer";
import Pokemon from "../../../Domain/Model/Pokemon";
import { Generation, generationLabels } from "../../../Domain/Model/GenerationInfo";
import { Link } from "react-router-dom";
import getRoute from "../../../Domain/Util/getRoute";
import UserContextService from "../../Service/UserContextService";

const HomeView = () => {
  const [email, setEmail] = useContext(UserContextService)

  const [currentGeneration, setCurrentGeneration] = useState<Generation>(generationLabels[0])
  const [searchPokemonName, setSearchPokemonName] = useState<string>('')
  const [pokemon, setPokemon] = useState<Pokemon | null>(null)
  const [showPokemonAltArt, setShowPokemonAltArt] = useState<boolean>(false)

  const onSearchPokemonChanged = useCallback(
    (name : string) => setSearchPokemonName(name.trim()),
    []
  )

  const onGenerationChanged = useCallback((generation: Generation) => {
      setSearchPokemonName('')
      setPokemon(null)
      setCurrentGeneration(generation)
    }, 
    [])

  return <React.Fragment>
    {(pokemon && showPokemonAltArt) && <PokemonModal name={pokemon.name} onPokemonArtClicked={() => setShowPokemonAltArt(false)}/>}
    {email === null && (<div className="pl-2 mb-4 text-sm">
      Want to log in?
      <Link to={getRoute('login')} className="
    ml-2 skew-y-3 inline-block p-1
    bg-orange-100 hover:bg-orange-200
    text-orange-600
    dark:bg-orange-700 dark:hover:bg-orange-600 dark:text-orange-200
    " style={{cursor: 'pointer'}}>
        <div className="-skew-y-3">Click here</div>
      </Link>
    </div>)}
    <div className='pl-2 mb-4 flex flex-row'>
      <div className='basis-3/4'>
        <div className="flex flex-row">
          <input
            data-test-id='search'
            type="search"
             className='
               appearance-none
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             '
             placeholder={'Start typing a Pokémon name...'} value={searchPokemonName}
             onChange={(e) => {
               onSearchPokemonChanged(e.target.value)
             }}/>
          {searchPokemonName.length !== 0 && <span
            data-test-id='clearSearchPokemonName'
            className='text-orange-500 font-mono dark:text-gray-200 ml-[-1.25rem] self-center'
            style={{cursor: 'pointer'}}
            onClick={() => onSearchPokemonChanged('')}
          >x</span>}
        </div>
      </div>
      <div className="basis-1/4 flex flex-row place-content-end pr-2">
        <label htmlFor="pokemon-gen" className="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\00a0gen.'] before:content-['Gen.']
        "/>
        <select id='pokemon-gen' className="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      " onChange={(e) => onGenerationChanged(e.target.value as Generation)}>
          {generationLabels.map((generationLabel) => <option key={generationLabel} value={generationLabel}>{generationLabel}</option>)}
        </select>
      </div>
    </div>
    <div className='mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]'>
      <div className='basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll'>
        <LoadingContainer>
          <PokemonList
            searchPokemonName={searchPokemonName}
            onPokemonClicked={(pokemon : Pokemon) => setPokemon(pokemon)}
            generation={currentGeneration}
          />
        </LoadingContainer>
      </div>

      {pokemon && <div className='basis-1/3 mb-4 md:mb-0 mx-2'>
        <LoadingContainer>
          <PokemonDetail pokemon={pokemon} callback={{
            onPokemonArtClicked: () => setShowPokemonAltArt(true),
            onCloseClicked: () => setPokemon(null)
          }}/>
        </LoadingContainer>
      </div>}
    </div>
  </React.Fragment>
};

export default hot(HomeView);
