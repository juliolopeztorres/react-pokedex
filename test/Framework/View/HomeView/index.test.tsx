import React from 'react'
import { createRoot, Root } from "react-dom/client"
import { act, fireEvent } from "@testing-library/react";
import { PokemonDetailCallback } from "../../../../src/Framework/View/HomeView/PokemonDetail";
import Pokemon from "../../../../src/Domain/Model/Pokemon";
import { Generation } from "../../../../src/Domain/Model/GenerationInfo";

jest.mock('../../../../src/Framework/View/Component/LoadingContainer', () => {
  return ({children}: {children: React.ReactElement}) => children
})

let outerOnPokemonClicked: (pokemon: Pokemon) => void

jest.mock('../../../../src/Framework/View/HomeView/PokemonList', () => {
  return ({searchPokemonName, onPokemonClicked, generation}: {searchPokemonName: string, onPokemonClicked: () => void, generation: Generation}) => {

    outerOnPokemonClicked = onPokemonClicked

    return <div>
      Pokemon List
      <span>Pokemon search name: {searchPokemonName}</span>
      <span>Pokemon generation: {generation}</span>
    </div>
  }
})

let outerOnPokemonArtCloseClicked: () => void

jest.mock('../../../../src/Framework/View/HomeView/PokemonModal', () => {
  return ({name, onPokemonArtClicked}: {name: string, onPokemonArtClicked: () => void}) => {

    outerOnPokemonArtCloseClicked = onPokemonArtClicked

    return <div>
      Pokemon Modal
      Name: {name}
    </div>
  }
})

let outerPokemonDetailCallback: PokemonDetailCallback

jest.mock('../../../../src/Framework/View/HomeView/PokemonDetail', () => {
  return ({pokemon, callback}: {pokemon: Pokemon, callback: PokemonDetailCallback}) => {

    outerPokemonDetailCallback = callback

    return <div>
      Pokemon Detail

      Pokemon Info: #{pokemon.id} {pokemon.name}
    </div>
  }
})

import HomeView from "../../../../src/Framework/View/HomeView";

let root: Root
let container: HTMLElement
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);

  root = createRoot(container)
});

afterEach(() => {
  // cleanup on exiting
  act(() => root.unmount())

  container.remove();
});

it('should render', () => {
  // Initial state
  act(() => {root.render(<HomeView/>)})

  expect(container).toMatchSnapshot()

  const search: HTMLInputElement | null = container.querySelector('input[data-test-id="search"]')
  expect(search).toBeTruthy()

  // Trigger filtering pokemon by name
  act(() => {fireEvent.input(search!, {target: {value: 'chari'}})})

  expect(container).toMatchSnapshot()

  // Trigger clean search input
  const cleanSearch: HTMLInputElement | null = container.querySelector('span[data-test-id="clearSearchPokemonName"]')
  expect(cleanSearch).toBeTruthy()

  act(() => {cleanSearch!.dispatchEvent(new MouseEvent('click', {bubbles: true}))})

  expect(container).toMatchSnapshot()

  // Trigger filtering pokemon by name again
  act(() => {fireEvent.input(search!, {target: {value: 'chari'}})})

  expect(container).toMatchSnapshot()

  // Select new generation
  const select: HTMLInputElement | null = container.querySelector('select#pokemon-gen')
  expect(select).toBeTruthy()

  act(() => {fireEvent.change(select!, {target: {value: '2nd'}})})

  expect(container).toMatchSnapshot()

  // Click on one pokemon from within the list
  act(() => {outerOnPokemonClicked(new Pokemon(1, 'charizard'))})

  expect(container).toMatchSnapshot()

  // Click on the pokemon art to see alternative one in modal
  act(() => {outerPokemonDetailCallback.onPokemonArtClicked()})

  expect(container).toMatchSnapshot()

  // Click on close the alternative art
  act(() => {outerOnPokemonArtCloseClicked()})

  expect(container).toMatchSnapshot()

  // Click on close pokemon detail
  act(() => {outerPokemonDetailCallback.onCloseClicked()})

  expect(container).toMatchSnapshot()
})
