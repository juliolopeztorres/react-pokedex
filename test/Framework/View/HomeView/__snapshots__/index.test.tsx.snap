// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          1st
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 2`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value="chari"
        />
        <span
          class="text-orange-500 font-mono dark:text-gray-200 ml-[-1.25rem] self-center"
          data-test-id="clearSearchPokemonName"
          style="cursor: pointer;"
        >
          x
        </span>
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
          chari
        </span>
        <span>
          Pokemon generation: 
          1st
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 3`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          1st
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 4`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value="chari"
        />
        <span
          class="text-orange-500 font-mono dark:text-gray-200 ml-[-1.25rem] self-center"
          data-test-id="clearSearchPokemonName"
          style="cursor: pointer;"
        >
          x
        </span>
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
          chari
        </span>
        <span>
          Pokemon generation: 
          1st
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 5`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          2nd
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 6`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          2nd
        </span>
      </div>
    </div>
    <div
      class="basis-1/3 mb-4 md:mb-0 mx-2"
    >
      <div>
        Pokemon Detail Pokemon Info: #
        1
         
        charizard
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 7`] = `
<div>
  <div>
    Pokemon Modal Name: 
    charizard
  </div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          2nd
        </span>
      </div>
    </div>
    <div
      class="basis-1/3 mb-4 md:mb-0 mx-2"
    >
      <div>
        Pokemon Detail Pokemon Info: #
        1
         
        charizard
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 8`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          2nd
        </span>
      </div>
    </div>
    <div
      class="basis-1/3 mb-4 md:mb-0 mx-2"
    >
      <div>
        Pokemon Detail Pokemon Info: #
        1
         
        charizard
      </div>
    </div>
  </div>
</div>
`;

exports[`should render 9`] = `
<div>
  <div
    class="pl-2 mb-4 flex flex-row"
  >
    <div
      class="basis-3/4"
    >
      <div
        class="flex flex-row"
      >
        <input
          class="
               pr-6
               w-[12rem]
               md:w-[20rem]
               pl-2
               py-1
               md:pl-6
               truncate
               focus:shadow-md dark:shadow-gray-700
               border
               border-red-200
               rounded-lg hover:border-red-400 focus:outline-none
               placeholder:text-red-200
               text-red-400
               dark:bg-slate-600
               dark:placeholder:text-gray-200
               dark:text-gray-200
               dark:border-red-400
             "
          data-test-id="search"
          placeholder="Start typing a Pokémon name..."
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="basis-1/4 flex flex-row place-content-end pr-2"
    >
      <label
        class="
        self-center mr-1
        text-sm font-mono text-slate-800 dark:text-gray-200
        before:md:content-['Pokémon\\00a0gen.'] before:content-['Gen.']
        "
        for="pokemon-gen"
      />
      <select
        class="
        text-end
        px-2
      bg-white
      border border-red-200 text-red-400 text-sm rounded-lg focus:outline-none
      hover:border-red-400
      dark:bg-slate-600 dark:border-red-400
      dark:text-gray-200
      "
        id="pokemon-gen"
      >
        <option
          value="1st"
        >
          1st
        </option>
        <option
          value="2nd"
        >
          2nd
        </option>
        <option
          value="3rd"
        >
          3rd
        </option>
        <option
          value="4th"
        >
          4th
        </option>
        <option
          value="5th"
        >
          5th
        </option>
        <option
          value="6th"
        >
          6th
        </option>
        <option
          value="7th"
        >
          7th
        </option>
        <option
          value="8th"
        >
          8th
        </option>
        <option
          value="9th"
        >
          9th
        </option>
      </select>
    </div>
  </div>
  <div
    class="mb-4 flex md:flex-row flex-col-reverse md:max-h-[40rem]"
  >
    <div
      class="basis-2/3 mx-2 p-4 border rounded-md border-orange-400 shadow dark:shadow-gray-700 overflow-y-scroll"
    >
      <div>
        Pokemon List
        <span>
          Pokemon search name: 
        </span>
        <span>
          Pokemon generation: 
          2nd
        </span>
      </div>
    </div>
  </div>
</div>
`;
